spec/Section 6 -- Execution.md

```diff
@@ -322,39 +312,315 @@ Unsubscribe(responseStream):
 
 - Cancel {responseStream}.
 
-## Executing Selection Sets
+## Executing the Root Selection Set
+
+To execute the root selection set, the object value being evaluated and the
+object type need to be known, as well as whether it must be executed serially,
+or may be executed in parallel.
+
+Executing the root selection set works similarly for queries (parallel),
+mutations (serial), and subscriptions (where it is executed for each event in
+the underlying Source Stream).
 
-To execute a _selection set_, the object value being evaluated and the object
+First, the selection set is turned into a grouped field set; then, we execute
+this grouped field set and return the resulting {data} and {errors}.
+
+ExecuteRootSelectionSet(variableValues, initialValue, objectType, selectionSet,
+serial):
+
+- If {serial} is not provided, initialize it to {false}.
+- Let {groupedFieldSet} and {newDeferUsages} be the result of
+  {CollectFields(objectType, selectionSet, variableValues)}.
+- Let {executionPlan} be the result of {BuildExecutionPlan(groupedFieldSet)}.
+- Let {data} and {incrementalDataRecords} be the result of
+  {ExecuteExecutionPlan(newDeferUsages, executionPlan, objectType, initialValue,
+  variableValues, serial)}.
+- Let {errors} be the list of all _field error_ raised while executing the
+  execution plan.
+- If {incrementalDataRecords} is empty, return an unordered map containing
+  {data} and {errors}.
+- Let {incrementalResults} be the result of {YieldIncrementalResults(data,
+  errors, incrementalDataRecords)}.
+- Wait for the first result in {incrementalResults} to be available.
+- Let {initialResult} be that result.
+- Return {initialResult} and {BatchIncrementalResults(incrementalResults)}.
+
+Note: {ExecuteExecutionPlan()} does not directly raise field errors from the
+incremental portion of the Execution Plan.
+
+### Yielding Incremental Results
+
+The procedure for yielding incremental results is specified by the
+{YieldIncrementalResults()} algorithm.
+
+YieldIncrementalResults(data, errors, incrementalDataRecords):
```

Comment:
Add a brief description to each function in this section

Reply:
should be completed by d9014e4